src/lib/hosts-file.ts

```diff
@@ -0,0 +1,123 @@
+import fs from 'fs';
+import { platform } from 'os';
+import { promisify } from 'util';
+import sudo from 'sudo-prompt';
+
+const readFile = promisify( fs.readFile );
+const writeFile = promisify( fs.writeFile );
+const sudoExec = promisify( sudo.exec );
+
+// Host file paths for different operating systems
+const HOST_FILES: Record< string, string > = {
+	win32: 'C:\\Windows\\System32\\drivers\\etc\\hosts',
+	darwin: '/etc/hosts',
+	linux: '/etc/hosts',
+};
+
+// Get the appropriate hosts file path for the current platform
+const getHostsFilePath = (): string => {
+	const currentPlatform = platform();
+	const hostsPath = HOST_FILES[ currentPlatform ];
+
+	if ( ! hostsPath ) {
+		throw new Error( `Unsupported platform: ${ currentPlatform }` );
+	}
+
+	return hostsPath;
+};
+
+/**
+ * Reads the hosts file content
+ */
+export const readHostsFile = async (): Promise< string > => {
+	try {
+		const hostsPath = getHostsFilePath();
+		const content = await readFile( hostsPath, 'utf8' );
+		return content;
+	} catch ( error ) {
+		console.error( 'Error reading hosts file:', error );
+		throw error;
+	}
+};
+
+/**
+ * Writes content to the hosts file (requires elevated permissions)
+ */
+export const writeHostsFile = async ( content: string ): Promise< void > => {
+	const hostsPath = getHostsFilePath();
+	const currentPlatform = platform();
+
+	try {
+		// On Windows and macOS, we need to use sudo to write to the hosts file
+		if ( currentPlatform === 'win32' || currentPlatform === 'darwin' ) {
+			const tempPath = '/tmp/wp-studio-hosts';
+			await writeFile( tempPath, content );
+			// @ts-expect-error promisify doesn't seem typed properly.
+			await sudoExec( `cat ${ tempPath } > ${ hostsPath }`, {
+				name: 'WordPress Studio',
+			} );
+		} else {
+			// For Linux, we also need sudo
+			const tempPath = '/tmp/wp-studio-hosts';
+			await writeFile( tempPath, content );
+			// @ts-expect-error promisify doesn't seem typed properly.
+			await sudoExec( `cat ${ tempPath } > ${ hostsPath }`, {
+				name: 'WordPress Studio',
+			} );
+		}
```

Comment:
Aren't these two branches identical?

Reply:
Looks like it

Reply:
I know in abstract that Windows has come a long way with WSL, but I have no idea if that means that it supports calling `cat` just like that. Now I wonder if the reason why there were two identical branches was that the model had the (correct?) instinct to treat write operations differently on Windows but in the end it didn't interpolate the right command invocation for it. Maybe...? Needs extra testing.